scheduler/test/e2e/scheduler_test.go

```diff
@@ -36,118 +43,165 @@ var _ = Describe("[E2E] KeptnScheduler", Ordered, func() {
-				workloadinstance.Status.PreDeploymentEvaluationStatus = "Succeeded"
-				err = k8sClient.Status().Update(ctx, workloadinstance)
+			It(" should be immediately scheduled", func() {
 
-				Expect(err).To(BeNil())
 				Eventually(func() error {
-					return podRunning(pod.Namespace, pod.Name)
-				}).Should(Succeed())
-
-				err = k8sClient.Delete(ctx, workloadinstance)
-				Expect(err).NotTo(HaveOccurred(), "could not remove workloadinstance")
+					return podScheduled(pod.Namespace, pod.Name)
+				}).WithTimeout(time.Second * 20).WithPolling(3 * time.Second).Should(Succeed())
 
 			})
 		})
 	})
 
-	Describe("If NOT annotated for keptn-scheduler", func() {
-		pause := imageutils.GetPauseImageName()
-		var (
-			pod *apiv1.Pod
-		)
+	Describe("If labeled for keptn-scheduler", func() {
+		labels1 := map[string]string{
+			K8sRecommendedWorkloadAnnotations: "myworkload",
+			K8sRecommendedVersionAnnotations:  "1.0.1",
+			K8sRecommendedAppAnnotations:      "mylabeledapp",
+		}
+		labels2 := map[string]string{
+			K8sRecommendedWorkloadAnnotations: "myworkload",
+			K8sRecommendedVersionAnnotations:  "1.0.2",
+			K8sRecommendedAppAnnotations:      "mylabeledapp",
+		}
+
+		pod1 := &apiv1.Pod{}
+		pod2 := &apiv1.Pod{}
 		BeforeEach(func() {
 			DeferCleanup(func() {
-				err := k8sClient.Delete(ctx, pod)
+				err := k8sClient.Delete(ctx, pod1)
+				logErrorIfPresent(err)
+				err = k8sClient.Delete(ctx, pod2)
 				logErrorIfPresent(err)
 			})
-
-			//create a test Pod
-			name := names.SimpleNameGenerator.GenerateName("my-testpod-")
-
-			pod = WithContainer(st.MakePod().
-				Namespace("default").
-				Name(name).
-				Req(map[apiv1.ResourceName]string{apiv1.ResourceMemory: "50"}).
-				ZeroTerminationGracePeriod().
-				Obj(), pause)
-
-			err := k8sClient.Create(ctx, pod)
-			Expect(ignoreAlreadyExists(err)).NotTo(HaveOccurred(), "could not add pod")
+			*pod1 = initPod(*pod1, nil, labels1, KeptnScheduler)
+			*pod2 = initPod(*pod2, nil, labels2, KeptnScheduler)
 		})
 
 		Context("a new Pod", func() {
+			It(" should stay pending if no workload instance is available", func() {
+				checkPending(pod1)
+			})
+			It(" should be scheduled when workload instance pre-evaluation checks are done", func() {
+				checkWorkload("mylabeledapp-myworkload-1.0.1", *pod1, "Succeeded")
+			})
 
-			It(" should be immediately scheduled", func() {
-
-				Eventually(func() error {
-					return podRunning(pod.Namespace, pod.Name)
-				}).Should(Succeed())
-
+			It(" should NOT be scheduled when workload instance pre-evaluation checks fails", func() {
+				checkWorkload("mylabeledapp-myworkload-1.0.2", *pod2, "Failed")
 			})
 		})
 	})
+
 })
 
-func initWorkloadInstance() *testv1alpha2.KeptnWorkloadInstance {
+func initPod(pod apiv1.Pod, annotations map[string]string, labels map[string]string, scheduler string) apiv1.Pod {
+
+	//create a test Pod
+	name := names.SimpleNameGenerator.GenerateName("my-testpod-")
+	pause := imageutils.GetPauseImageName()
```

Comment:
to avoid confusion by reading `pause` as the action
```suggestion
	pauseImg := imageutils.GetPauseImageName()
```